src/adapter/src/catalog/builtin_table_updates.rs

```diff
@@ -745,4 +745,42 @@ impl CatalogState {
             diff,
         }
     }
+
+    pub fn pack_storage_usage_update(
+        &self,
+        event_details: &EventDetails,
+    ) -> Result<BuiltinTableUpdate, Error> {
+        let event_details = Jsonb::from_serde_json(event_details.as_json())
```

Comment:
This should be a slice of datums. Use the same form as in the other functions. For example:

```
        let row = Row::pack_slice(&[
            Datum::Int64(id.try_into().expect("Replica IDs should not overflow i64")),
            Datum::TimestampTz(last_heartbeat),
        ]);
```